src/daft-table/src/growable/mod.rs

```diff
@@ -0,0 +1,67 @@
+use common_error::DaftResult;
+use daft_core::{
+    array::growable::{make_growable, Growable},
+    Series,
+};
+
+use crate::Table;
+
+pub struct GrowableTable<'a> {
+    growables: Vec<Box<dyn Growable + 'a>>,
+}
+
+impl<'a> GrowableTable<'a> {
+    pub fn new(tables: &[&'a Table], use_validity: bool, capacity: usize) -> Self {
+        let num_tables = tables.len();
+        assert!(num_tables > 0);
+        let first = tables.first().unwrap();
+        let num_columns = first.num_columns();
+        let schema = &first.schema;
+        let mut series_list = (0..num_columns)
+            .map(|_| Vec::<&Series>::with_capacity(num_columns))
```

Comment:
Should the capacity here be `num_tables` instead? If i understand correctly, `series_list` is a list of lists, each list containing all of the columns for a given field for all tables.